src/server/middleware/access-tokens/TokenServer.js

```diff
@@ -0,0 +1,104 @@
+/*eslint-env node*/
+
+/**
+ * @author lattmann / https://github.com/lattmann
+ * @author ksmyth / https://github.com/ksmyth
+ * @author pmeijer / https://github.com/pmeijer
+ *
+ curl http://localhost:8855/rest/executor/info/77704f10a36aa4214f5b0095ba8099e729a10f46
+ curl -X POST -H "Content-Type: application/json"
+ -d {} http://localhost:8855/rest/executor/create/77704f10a36aa4214f5b0095ba8099e729a10f46
+ curl -X POST -H "Content-Type: application/json"
+ -d {\"status\":\"CREATED\"} http://localhost:8855/rest/executor/update/77704f10a36aa4214f5b0095ba8099e729a10f46
+ */
+
+'use strict';
+
+var express = require('express'),
+    Chance = require('chance'),
+    // Mongo collections
+    TOKEN_LIST = '_tokenList';
+
+/**
+ *
+ * @param {object} options - middlewareOptions
+ * @param {GmeLogger} options.logger - logger to fork off from
+ * @param {GmeConfig} options.gmeConfig - gmeConfig
+ * @param {function} options.ensureAuthenticated
+ * @constructor
+ * @ignore
+ */
+function TokenServer(options) {
+    var self = this,
+        router = express.Router();
+
+    self.logger = options.logger.fork('middleware:TokenServer');
+    self.ensureAuthenticated = options.ensureAuthenticated;
+    self.getUserId = options.getUserId;
+    self.router = router;
+    self.tokens = new AccessTokens();
```

Comment:
either I see differently or the tokens are stored in a common place?
I thought we store them by user (though that might make using them harder - but we can store the userid inside...)

Reply:
These are the personal access tokens (completely separate from the jwt webgme tokens used for user authentication). These are currently stored under the `_tokenList` collection w/ the userId as one of the fields. The `AccessTokens` class is created only to separate the business logic of token manipulation from the REST API.

Reply:
ok, lets leave it like this.